data_safe_haven/resources/dns_server/AdGuardHome.mustache.yaml

```diff
@@ -0,0 +1,25 @@
+users:
+  - name: {{admin_username}}
+    password: {{admin_password_encrypted}}
+dns:
+  blocking_mode: nxdomain
+  upstream_dns:
+    - "{{upstream_dns}}"
+  upstream_dns_file: ""
+  bootstrap_dns:
+    - "{{upstream_dns}}"
+querylog:
+  enabled: true
+filters:
+user_rules:
+  {{#filter_deny}}
+  - "{{.}}"
+  {{/filter_deny}}
```

Comment:
Given that we want to block everything except the allowed domains, should we just hard code here?

Reply:
Good point, I originally had this as a standalone Docker application where you might want to specify, but you're right that there's no need to do so here.